docs/check/overview.md

```diff
@@ -0,0 +1,583 @@
+```
+model
+  schema 1.1
+
+type user
+
+type folder
+ relations
+   define viewer: [user]
+
+type document
+  relations
+    define parent: [folder]
+    define editor: [user] or editor from parent
+    define viewer: editor
+```
+The `document#viewer` relation is "rewritten" indirectly as `document#editor`, which means that in order to evaluate a specific relationship of the form `Check(document:1#viewer@user:jon)` we must actually rewrite that lookup as `Check(document:1#editor@user:jon)`. Similarly, the `document#editor` relationship is rewritten using a set union (e.g. `or` keyword) such that if a user is either directly an editor of the document or an editor on the parent folder, then the user can edit the document. This kind of rewritten behavior allows us to model hierarchical semantics. These "rewrites" make up the base-line behavior behind the semantics of the FGA graph traversal, and we explain each of them with examples in further detail below.
+
+## Definitions
+Familiarity with basic [OpenFGA Concepts](https://openfga.dev/docs/concepts) is expected, but the following are some terms used throughout this document that are more specific to the content herein:
+
+* **Relation Rewrite (Rewrite)** - a rule that expresses an FGA relationship as a composition of other relationships. For example, you can "rewrite" a viewer relationship as editor or express a hierarchy between two objects. These rules can also be composed using set operations including union, intersection, and exclusion.
+
+* **Rewrite Operand** - for rewrites involving set operations (e.g. union, intersection, exclusion) a rewrite operand refers to the relation rewrites that are evaluated as part of the set expression. For example, the rewrite operands of the definition `a or b` are `a` and `b`.
+
+* **Expansion** - refers to the <u>process</u> of iteratively evaluating one or more subproblems in order to determine some FGA query outcome. 
+
+* **Subproblem** - an FGA query that is contingent on one or more evaluations of nested relationship evaluations (including itself).
+
+* **Dispatch** - refers to the process of evaluating a nested subproblem in order to resolve some parent subproblem.
+
+## Direct Rewrites
+Direct rewrites establish the most basic form of FGA relationship rewrite definitions. These direct rewrites allow developers to express the most basic of authorization model semantics, including direct permissions, indirect or computed permissions, and/or hierarchical permissions.
+
+> ℹ️ The term "permission" is used above, but from here forward we will formally refer to permissions simply as relationships. A users/subject has a permission if they have the relationship, and a relationship can be realized through one or more relationship rewrites.
+
+### Direct Relationships
```

Comment:
```suggestion
### Model with Direct Relationships. Example Check
```

here and elsewhere. I skipped through all these thinking that they were going to explain what these terms meant, only to realize that these actually have examples of Check resolutions :)

Reply:
We do explain the term "Direct Relationships" below. Specifically, we say
> If a relation can be directly assigned or given/shared with a specific user/subject, then we refer to it as a "direct relationship". Direct relationships are any of those relationships that, for a particular authorization model, can be written to an [FGA Store](https://openfga.dev/docs/concepts#what-is-a-store) using the [Write API](https://openfga.dev/api/service#/Relationship%20Tuples/Write).

The content of the doc is organized by rewrite rule categories. There are 2 categories - direct rewrite rules and set rewrite rules. The set rewrite rules compose other set rewrite rules and/or direct rewrite rules. Self (e.g. direct relationships), computed userset, and TTU encompass the direct rewrite rules. Union, intersection, and exclusion encompass the set rewrite rules.

The purpose of this section was to a) outline the direct rewrite rules and what they do, specifically in this subsection for direct relationships and b) provide examples of them. I feel like the content as is adequately portrays the intent.